pkgs/by-name/li/libtapi/package.nix

```diff
@@ -0,0 +1,180 @@
+    # `fetchgit` can’t be used in the Darwin bootstrap.
+    (fetchFromGitHub {
+      name = "apple-llvm-src";
+      owner = "apple";
+      repo = "llvm-project";
+      inherit (appleLlvm) rev hash;
+    })
+  ];
+
+  patches = [
+    # Older versions of ld64 may not support `-no_exported_symbols`, so use it only
+    # when the linker supports it.
+    # Note: This can be dropped once the bootstrap tools are updated after the ld64 update.
+    ./0001-Check-for-no_exported_symbols-linker-support.patch
+    # The recommended upstream revision of Apple’s LLVM fork needs this patch, or
+    # `tapi stubify` will crash when generating stubs.
+    (fetchpatch {
+      url = "https://github.com/apple/llvm-project/commit/455bf3d1ccd6a52df5e38103532c1b8f49924edc.patch";
+      hash = "sha256-ujZcfdAls20JPIvjvO2Xv8st8cNTY/XTEQusICKBKSA";
+    })
+    # Updates `JSONReaderWriter` to work with the API change in the above patch.
+    ./0002-Pass-fileType-to-writeToStream.patch
+    # Fix build on Linux. GCC is more picky than clang about the field order.
+    ./0003-Match-designator-order-with-declaration-order.patch
+  ];
+
+  postPatch =
+    ''
+      # Enable building on non-Darwin platforms
+      substituteInPlace tapi/CMakeLists.txt \
+        --replace-fail 'message(FATAL_ERROR "Unsupported configuration.")' ""
+
+      # Remove the client limitation on linking to libtapi.dylib.
+      substituteInPlace tapi/tools/libtapi/CMakeLists.txt \
+        --replace-fail '-allowable_client ld' ""
+      # Replace hard-coded installation paths with standard ones.
+      declare -A installdirs=(
+        [bin]=BINDIR
+        [include]=INCLUDEDIR
+        [lib]=LIBDIR
+        [local/bin]=BINDIR
+        [local/share/man]=MANDIR
+        [share/man]=MANDIR
+      )
+      for dir in "''${!installdirs[@]}"; do
+        cmakevar=CMAKE_INSTALL_''${installdirs[$dir]}
+        for cmakelist in $(grep -rl "DESTINATION $dir" tapi); do
+          substituteInPlace "$cmakelist" \
+            --replace-fail "DESTINATION $dir" "DESTINATION \''${$cmakevar}"
+        done
+      done
+      # Doesn’t seem to exist publically.
+      substituteInPlace tapi/test/CMakeLists.txt \
+        --replace-fail tapi-configs ""
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      # Remove Darwin-specific versioning flags.
+      substituteInPlace tapi/tools/libtapi/CMakeLists.txt \
+          --replace-fail '-current_version ''${DYLIB_VERSION} -compatibility_version 1' ""
+    '';
+
+  preUnpack = ''
+    mkdir source
+  '';
+
+  sourceRoot = "source";
+
+  postUnpack = ''
+    chmod -R u+w apple-llvm-src tapi-src
+    mv apple-llvm-src/{clang,cmake,llvm,utils} source
+    mv tapi-src source/tapi
+  '';
+
+  strictDeps = true;
+
+  buildInputs = [ zlib ]; # Upstream links against zlib in their distribution.
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    python3
+  ];
+
+  cmakeDir = "../llvm";
```

Comment:
Does this not refer to `source/../llvm`, which shouldn't exist?

Reply:
It actually relative to the build directory which is `source/build`, so `cmakeDir` resolves to `source/llvm`.